Csv/CsvReader.cs

```diff
@@ -135,9 +135,23 @@ private static IEnumerable<ICsvLine> ReadImpl(TextReader reader, CsvOptions opti
                         continue;
                 }
 
-                // TODO: #11 might need to read another line if this one isn't ended yet (i.e. open quoted value)
+                var record = new ReadLine(headers, headerLookup, index, line, options);
```

Comment:
I finally realized that unless you fully split the line, you have no way of truly knowing whether it is an unterminated quote-enclosed value.  So I slightly modified the ReadLine class to allow me to access the split but untrimmed values so that I could check if we need to read another line (and reconstruct the ReadLine for additional verification).  Once I've verified that the ReadLine object is valid (not containing an unterminated final value), then I can return it back to the method caller.